phpunit/class-gutenberg-rest-navigation-fallback-controller-test.php

```diff
@@ -158,10 +158,39 @@ public function test_prepare_item() {
 	}
 
 	/**
-	 * @doesNotPerformAssertions
+	 * Tests that the correct filters are applied to the context parameter.
+	 *
+	 * The REST response for the Posts Controller doesn't return all fields,
+	 * when the context is set to embed. We need to add additional fields
+	 * for the navigation fallback, so that when it embeds a navigation,
+	 * the required fields are present.
+	 *
+	 * @covers wp_add_fields_to_navigation_fallback_embeded_links
 	 */
 	public function test_context_param() {
-		// Covered by the core test.
+		// First we'll use the navigation fallback to get a link to the navigation endpoint.
+		$request  = new WP_REST_Request( 'GET', '/wp-block-editor/v1/navigation-fallback' );
+		$response = rest_get_server()->dispatch( $request );
+		$links = $response->get_links();
+
+		// Extract the navigation endpoint URL from the response.
+		$embedded_navigation_href = $links['self'][0]['href'];
+		preg_match('/\?rest_route=(.*)/', $embedded_navigation_href, $matches );
+		$navigation_endpoint = $matches[1];
+
+		// Fetch a navigation from the endpoint, with the context parameter set to embed.
+		$request  = new WP_REST_Request( 'GET', $navigation_endpoint );
+		$request->set_param( 'context', 'embed' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		// Verify that the additional fields are present.
```

Comment:
Nit: can we collocate the subfield checks for fields of the same type (e.g. content, title...etc)?